storage/badger/operation/dkg.go

```diff
@@ -0,0 +1,52 @@
+package operation
+
+import (
+	"errors"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/dkg"
+	"github.com/onflow/flow-go/storage"
+)
+
+// InsertMyDKGPrivateInfo stores the random beacon private key for the given epoch.
+//
+// CAUTION: This method stores confidential information and should only be
+// used in the context of the secrets database. This is enforced in the above
+// layer (see storage.DKGKeys).
+func InsertMyDKGPrivateInfo(epochCounter uint64, info *dkg.DKGParticipantPriv) func(*badger.Txn) error {
+	return insert(makePrefix(codeDKGPrivateInfo, epochCounter), info)
+}
+
+// RetrieveMyDKGPrivateInfo retrieves the random beacon private key for the given epoch.
+//
+// CAUTION: This method stores confidential information and should only be
+// used in the context of the secrets database. This is enforced in the above
+// layer (see storage.DKGKeys).
+func RetrieveMyDKGPrivateInfo(epochCounter uint64, info *dkg.DKGParticipantPriv) func(*badger.Txn) error {
+	return retrieve(makePrefix(codeDKGPrivateInfo, epochCounter), info)
+}
+
+// InsertDKGStartedForEpoch stores a flag indicating that the DKG has been started for
+// the given epoch.
+func InsertDKGStartedForEpoch(epochCounter uint64) func(*badger.Txn) error {
+	return insert(makePrefix(codeDKGStarted, epochCounter), true)
```

Comment:
wonder if it's useful to store a block ID of when we started the DKG instead of just a `bool`, since the bool value is always `true`.

when we joining the network, we will retrieve the block ID back, would it be useful to do a comparison with the snapshot's block ID and the stored block ID when DKG was started?

Reply:
I don't really see what we would use this for to be honest. 

> when we joining the network, we will retrieve the block ID back

When joining the network this flag wouldn't be set, because we would not have already started the DKG. If we store a block ID here, it will always be either:
* the block at which the `EpochSetupPhaseStarted` protocol event is emitted, the majority of the time
* the finalized head of the root snapshot used to bootstrap the node, when joining the network at an epoch boundary

I don't think we have a need to know which of these cases occurred.